### ISSUE TITLE
PUT copy

### ISSUE THREAD

**Issue by kuenishi** [basho/riak_cs#363]
s3fs is using x-amz-copy-source to update metadata.

**Comment by slfritchie** [comment]
For example, to mimic what s3fs is doing
1. Delete the thing, e.g. bucket `test` object `foo`
   
   s3cmd del s3://test/foo
2. Create the thing with some metadata:
   
   s3curl.pl --debug --put ~/foo --id=$MY_AWS_KEY --key=$MY_AWS_SECRET --contentType "text/plain" -- -H "x-amz-meta-custom1: foo0" -H "x-amz-meta-custom2: bar0" -x localhost:8080 http://s3.amazonaws.com/test/foo -v
3. Update the thing with new metadata, like s3fs does:
   
   s3curl.pl --debug --put /tmp/zero --id=$MY_AWS_KEY --key=$MY_AWS_SECRET --contentType "text/plain" -- -H "x-amz-meta-custom1: foo2222" -H "x-amz-meta-custom2: bar2222" -H "x-amz-meta-custom3: bazbazbaz" -H "x-amz-copy-source: /test/foo" -x localhost:8080 http://s3.amazonaws.com/test/foo -v

Output looks like:

```
s3curl: Found the url: host=s3.amazonaws.com; port=; uri=/test/foo; query=;
s3curl: ordinary endpoint signing case
s3curl: StringToSign='PUT\n\ntext/plain\nMon, 18 Feb 2013 16:38:49 GMT\nx-amz-copy-source:/test/foo\nx-amz-meta-custom1:foo2222\nx-amz-meta-custom2:bar2222\nx-amz-meta-custom3:bazbazbaz\n/test/foo'
s3curl: signature='j7OXwFdDif/hAEz54Bj9uh9SVnk='
s3curl: exec curl -H Date: Mon, 18 Feb 2013 16:38:49 GMT -H Authorization: AWS J2IP6WGUQ_FNGIAN9AFI:j7OXwFdDif/hAEz54Bj9uh9SVnk= -L -H content-type: text/plain -T /tmp/zero -H x-amz-meta-custom1: foo2222 -H x-amz-meta-custom2: bar2222 -H x-amz-meta-custom3: bazbazbaz -H x-amz-copy-source: /test/foo -x localhost:8080 http://s3.amazonaws.com/test/foo -v
* About to connect() to proxy localhost port 8080 (#0)
*   Trying ::1...
* Connection refused
*   Trying 127.0.0.1...
* connected
* Connected to localhost (127.0.0.1) port 8080 (#0)
> PUT http://s3.amazonaws.com/test/foo HTTP/1.1
> User-Agent: curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8r zlib/1.2.5
> Host: s3.amazonaws.com
> Accept: */*
> Proxy-Connection: Keep-Alive
> Date: Mon, 18 Feb 2013 16:38:49 GMT
> Authorization: AWS J2IP6WGUQ_FNGIAN9AFI:j7OXwFdDif/hAEz54Bj9uh9SVnk=
> content-type: text/plain
> x-amz-meta-custom1: foo2222
> x-amz-meta-custom2: bar2222
> x-amz-meta-custom3: bazbazbaz
> x-amz-copy-source: /test/foo
> Content-Length: 0
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 200 OK
< Server: Riak CS
< ETag: "d41d8cd98f00b204e9800998ecf8427e"
< Date: Mon, 18 Feb 2013 16:38:49 GMT
< Content-Type: text/plain
< Content-Length: 0
< 
* Connection #0 to host localhost left intact
* Closing connection #0
```
1. Check the metadata attributes (new ones are present) and file size (bummer, content-length is zero):
   
   s3curl.pl --id=$MY_AWS_KEY --key=$MY_AWS_SECRET -- -x localhost:8080 http://s3.amazonaws.com/test/foo -v | & egrep -i 'amz|length'

Output says:

```
< x-amz-meta-custom3: bazbazbaz
< x-amz-meta-custom2: bar2222
< x-amz-meta-custom1: foo2222
< Content-Length: 0
```

**Comment by slfritchie** [comment]
Sorry, the Heya system got screwed up by my attempt to join two cards together failed.  This item is addressed by PR #425, closing.
